### PR TITLE
[Typescript] Worker improvements

### PR DESCRIPTION
**Pull request by dermasmid**
A couple of minor improvements to the TS sdk

- [update node image to 18 and use slim](https://github.com/apache/beam/pull/29903/commits/cdfb5f91b2216f488155a5f521289e4df39469ac)
- [allow user to log structured logs](https://github.com/apache/beam/pull/29903/commits/1c5de3f292b99308c62dff89844d1aebd8c24ea0)
- [get package name and version from pakcage.json at runtime](https://github.com/apache/beam/pull/29903/commits/8ef4e877f48672acdb7f908c0450b6cf0b12cee0)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by johnjcasey** [comment]
assign to next reviewer

**Comment by johnjcasey** [comment]
I'm not familiar with TS, so I'm not a good reviewer for this

**Comment by ahmedabu98** [comment]
assign to next reviewer

**Comment by ahmedabu98** [comment]
CC @robertwb who is more familiar with TS SDK

File: sdks/typescript/build.sh [hunk 1]
```diff
@@ -22,9 +22,6 @@
 
 set -e
 
-# Make the packaging version available to the code.
```
Comment (robertwb) [inline sdks/typescript/build.sh:25]: Thanks! I couldn't figure out how to do this myself.

File: sdks/typescript/build.sh [hunk 2]
```diff
@@ -22,9 +22,6 @@
...
-echo "export const version = \"$npm_package_version\";" > src/apache_beam/version.ts
-
 # Using npx to execute ttsc from the local node_modules environment.
 npx ttsc -p .
```
Comment (robertwb) [inline sdks/typescript/build.sh:26]: Now that we're down to a single command, how about removing build.sh and just putting this `ttsc` call in package.json directly?

**Review by robertwb — APPROVED** [review]

File: sdks/typescript/src/apache_beam/worker/worker_main.ts [hunk 3]
```diff
@@ -64,10 +64,10 @@ async function main() {
     },
     options,
   );
+  console.info("Worker started.");
```
[the remark this answers is not in the document]
Comment (dermasmid) [reply sdks/typescript/src/apache_beam/worker/worker_main.ts:67]: The await for the logs blocks, so I didn't get the started log

File: sdks/typescript/build.sh (hunk 2 above)
Comment (dermasmid) [reply sdks/typescript/build.sh:26]: I'll work on that

File: sdks/typescript/src/apache_beam/worker/worker_main.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/typescript/src/apache_beam/worker/worker_main.ts:67]: Oh, yes. I suppose the intent is to "start this in the background." Not sure the idiomatic way to do this, but the change is fine.

[the remark this answers is not in the document]
Comment (dermasmid) [reply sdks/typescript/src/apache_beam/worker/worker_main.ts:67]: We could remove the await on the logs

**Pull request merged**